helper.php

```diff
@@ -805,6 +821,42 @@ public function html_table() {
         return '<div class="table">' . $form->toHTML() . '</div>';
     }
 
+    /**
+     * Display tag cleaner
+     *
+     * @return string
+     */
+    public function html_clean()
+    {
+        $invalid = $this->getInvalidTaggings();
+
+        if (!$invalid) {
+            return '<p><strong>' . $this->getLang('admin no invalid') . '</strong></p>';
+        }
+
+        $form = new Form();
+        $form->setHiddenField('do', 'admin');
+        $form->setHiddenField('page', $this->getPluginName());
+        $form->addButton('cmd[clean]', $this->getLang('admin clean'));
+
+        $html = $form->toHTML();
+
+        $html .= '<div class="table"><table class="inline plugin_tagging">';
+        $html .= '<thead><tr><th>' .
+            $this->getLang('admin nonexistent page') .
+            '</th><th>' .
+            $this->getLang('admin tags') .
+            '</th></tr></thead><tbody>';
+
+        foreach ($invalid as $row) {
+            $html .= '<tr><td>' . $row['pid'] . '</td><td>' . $row['tags'] . '</td></tr>';
```

Comment:
do we need to html escape something here?

Reply:
No, we have only cleaned tags and page ids here.